aptos-move/e2e-move-tests/src/aggregator_v2.rs

```diff
@@ -189,267 +144,285 @@ pub fn init(
     )
 }
 
-pub fn check(
-    harness: &mut MoveHarness,
-    agg_loc: &AggLocation,
-    expected: u128,
-) -> SignedTransaction {
-    create_entry_agg_func_with_arg(harness, "0x1::aggregator_v2_test::check", agg_loc, expected)
-}
+impl AggV2TestHarness {
+    fn create_entry_agg_func_no_arg(
+        &mut self,
+        name: &str,
+        agg_loc: &AggLocation,
+    ) -> SignedTransaction {
+        self.txn_index += 1;
+        self.harness.create_entry_function(
+            &self.txn_accounts[self.txn_index % self.txn_accounts.len()],
+            str::parse(name).unwrap(),
+            vec![agg_loc.element_type.get_type_tag()],
+            vec![
+                bcs::to_bytes(&agg_loc.address).unwrap(),
+                bcs::to_bytes(&(agg_loc.use_type as u32)).unwrap(),
+                bcs::to_bytes(&agg_loc.index).unwrap(),
+            ],
+        )
+    }
 
-pub fn check_snapshot(
-    harness: &mut MoveHarness,
-    snap_loc: &AggLocation,
-    expected: u128,
-) -> SignedTransaction {
-    println!(
-        "Check snapshot argument: {:?}",
-        snap_loc.element_type.value_to_bcs(expected)
-    );
-    create_entry_agg_func_with_arg(
-        harness,
-        "0x1::aggregator_v2_test::check_snapshot",
-        snap_loc,
-        expected,
-    )
-}
+    fn create_entry_agg_func_with_arg(
```

Comment:
Why it is better to have _no_arg, _arg and _two_args functions over passing a vector of arguments?